modules/launcher/src/main/java/org/mule/runtime/module/launcher/ApplicationClassLoaderBuilderFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.launcher;
+
+import org.mule.runtime.module.launcher.domain.DomainRepository;
+
+/**
+ * Factory to create instances of {@code ApplicationClassLoaderBuilder}.
+ *
+ * @since 4.0
+ */
+public class ApplicationClassLoaderBuilderFactory
```

Comment:
Looks like there is a missing concept as ApplicationClassLoaderBuilderFactory and ArtifactClassLoaderBuilderFactory could share a common interface.